keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+
+            policies.add(new UserAgentPolicy(userAgent));
+            policies.add(new RequestIdPolicy());
+            policies.add(retryPolicy);
+            policies.add(new KeyvaultCredentialsPolicy(credentials, vaultEndPoint));
+
+            policies.addAll(this.policies);
+
+            policies.add(new HttpLoggingPolicy(httpLogDetailLevel));
+
+            HttpPipeline pipeline = httpClient == null
+                    ? new HttpPipeline(policies)
+                    : new HttpPipeline(httpClient, policies);
+
+            return new SecretClient(vaultEndPoint, pipeline);
+        }
+
+        /**
+         * Sets the credentials to use when authenticating HTTP requests.
+         *
+         * @param vaultEndPoint The vault endpoint url is used as destination on Azure to send requests to.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code credentials} is {@code null}.
+         */
+        public Builder vaultEndPoint(String vaultEndPoint) {
+            Objects.requireNonNull(vaultEndPoint);
+            this.vaultEndPoint = vaultEndPoint;
+            return this;
+        }
+
+        /**
+         * Sets the credentials to use when authenticating HTTP requests.
+         *
+         * @param credentials The credentials to use for authenticating HTTP requests.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code credentials} is {@code null}.
+         */
+        public Builder credentials(ServiceClientCredentials credentials) {
+            Objects.requireNonNull(credentials);
+            this.credentials = credentials;
+            return this;
+        }
+
+        /**
+         * Sets the logging level for HTTP requests and responses.
+         *
+         * @param logLevel The amount of logging output when sending and receiving HTTP requests/responses.
+         * @return The updated Builder object.
+         */
+        public Builder httpLogDetailLevel(HttpLogDetailLevel logLevel) {
+            httpLogDetailLevel = logLevel;
+            return this;
+        }
+
+        /**
+         * Adds a policy to the set of existing policies that are executed after
+         * {@link com.azure.keyvault.SecretClient} required policies.
+         *
+         * @param policy The retry policy for service requests.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code policy} is {@code null}.
+         */
+        public Builder addPolicy(HttpPipelinePolicy policy) {
+            Objects.requireNonNull(policy);
+            policies.add(policy);
+            return this;
+        }
+
+        /**
+         * Sets the HTTP client to use for sending and receiving requests to and from the service.
+         *
+         * @param client The HTTP client to use for requests.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code client} is {@code null}.
```

Comment:
No null check means no NPE will be thrown

Reply:
removed @throws